src/domain/entity/Entity.ts

```diff
@@ -8,20 +9,20 @@ export interface IEntityDTO {
     id: string;
 }
 
-const isEntity = (v: any): v is Entity<any> => v instanceof Entity;
-
 export abstract class Entity<T extends IEntityDTO> implements IEntity<T> {
 
-    protected readonly _id: IUniqueId<string>;
+    protected readonly _id: IIdentifier<string>;
     protected readonly props: Omit<T, 'id'>;
 
+    static isEntity(v: any): v is Entity<any> { return v instanceof Entity; }
+
     constructor({ id, ...props }: T) {
-        this._id = new UniqueId(id);
+        this._id = new UniqueID(id);
         this.props = props;
     };
 
     public equals(entity?: Entity<T>): boolean {
-        if (!entity && !isEntity(entity)) return false;
+        if (!entity && !Entity.isEntity(entity)) return false;
```

Comment:
Why do we need to separate checks here? What if we try to short this to one-liner
`return entity && Entity.isEntity(entity) && (this === entity || this._id.equals(entity._id))` 
I have doubts about accessing _id directly ('cause _id is protected and therefore should not be accessed directly, 'cause protected it's form of private). Maybe it would be better to create getter for id?

Reply:
Changed, thanks